store/registry_test.go

```diff
@@ -50,3 +67,221 @@ func Test_calculateShardID(t *testing.T) {
+		wantErr     bool
+	}{
+		{
+			name: "can calculate correctly for unsharded bot",
+			args: args{agentID: "0xbot1", scannerAddress: "0xscanner1"},
+			fields: fields{
+				mc: func(t *testing.T) *mock_manifest2.MockClient {
+					ctrl := gomock.NewController(t)
+					m := mock_manifest2.NewMockClient(ctrl)
+
+					m.EXPECT().GetAgentManifest(gomock.Any(), gomock.Any()).Return(
+						&manifest.SignedAgentManifest{
+							Manifest: &manifest.AgentManifest{ImageReference: &mockImageRef1},
+						}, nil,
+					).Times(2)
+					return m
+				},
+				rc: func(t *testing.T) *mock_registry.MockClient {
+					ctrl := gomock.NewController(t)
+					rc := mock_registry.NewMockClient(ctrl)
+					rc.EXPECT().GetAgent("0xbot1").Return(&registry.Agent{Manifest: "QmfFo3bw3QnuHEMPnsWmxHoHzNYd98vuEjGQJYVg9knYBm"}, nil)
+					rc.EXPECT().NumScannersForByChain(gomock.Any(), gomock.Any()).Return(big.NewInt(int64(mockAssigns1)), nil)
+					return rc
+				},
+				cfg: config.Config{},
+				mu:  sync.Mutex{},
+			},
+			wantErr:     false,
+			wantShards:  mockShards1,
+			wantTarget:  mockAssigns1,
+			wantShardID: 0,
+		},
+		{
+			name: "can calculate correctly for sharded bot",
+			args: args{agentID: "0xbot1", scannerAddress: "0xscanner1"},
+			fields: fields{
+				mc: func(t *testing.T) *mock_manifest2.MockClient {
+					ctrl := gomock.NewController(t)
+					m := mock_manifest2.NewMockClient(ctrl)
+
+					m.EXPECT().GetAgentManifest(gomock.Any(), gomock.Any()).Return(
+						&manifest.SignedAgentManifest{
+							Manifest: &manifest.AgentManifest{
+								ImageReference: &mockImageRef1, ChainSettings: map[string]manifest.AgentChainSettings{
+									keyDefaultChainSetting: {
+										Shards: mockShards2,
+									},
+								},
+							},
+						}, nil,
+					).Times(2)
+					return m
+				},
+				rc: func(t *testing.T) *mock_registry.MockClient {
+					ctrl := gomock.NewController(t)
+					rc := mock_registry.NewMockClient(ctrl)
+					rc.EXPECT().GetAgent("0xbot1").Return(&registry.Agent{Manifest: "QmfFo3bw3QnuHEMPnsWmxHoHzNYd98vuEjGQJYVg9knYBm"}, nil)
+					rc.EXPECT().NumScannersForByChain(gomock.Any(), gomock.Any()).Return(big.NewInt(int64(mockAssigns2)), nil)
+					rc.EXPECT().IndexOfAssignedScannerByChain(gomock.Any(), gomock.Any(), gomock.Any()).Return(mockScannerIdx, nil)
+					return rc
+				},
+				cfg: config.Config{},
+				mu:  sync.Mutex{},
+			},
+			wantErr:     false,
+			wantShards:  mockShards2,
+			wantTarget:  2,
+			wantShardID: 1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(
+			tt.name, func(t *testing.T) {
+				rs, _ := NewRegistryStoreFromClients(context.Background(), tt.fields.cfg, tt.fields.mc(t), tt.fields.rc(t))
+				gotShardID, gotShards, gotTarget, err := rs.FindScannerShardIDForBot(tt.args.agentID, tt.args.scannerAddress)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("FindScannerShardIDForBot() error = %v, wantErr %v", err, tt.wantErr)
+					return
+				}
+				if gotShardID != tt.wantShardID {
```

Comment:
Instead of these manual checks, can we use `require.Assertions`? We are able to make more types of assertions very simply using that. There are methods like `Error`, `NoError`, `True`, `False`, `Len`, `Equal`, `EqualError` etc. Here are some example tests that initializes and uses this:
https://github.com/forta-network/forta-core-go/blob/master/utils/ethutils/iterator_test.go#L50
https://github.com/forta-network/forta-core-go/blob/master/utils/slicemap/slice_map_test.go#L9